### ISSUE TITLE
Create native AngularJS directives for RichWidgets

### ISSUE THREAD

**Issue by VineetReynolds** [richwidgets/richwidgets#57]
Creating AngularJS directives would be necessary to ensure that RichWidgets can be used in AngularJS apps.

Note - it would be better to create native AngularJS widgets on the lines of the Angular-UI project.

While we can wrap jQuery/jQueryUI-like plugins as AngularJS directives, this is not recommended since it does not address the impedance mismatch between the Angular view of the DOM and the plugin's view of the DOM. Directives that wrap plugins require a lot of scope.$apply invocations to ensure that AngularJS is notified of DOM updates performed outside AngularJS.

This might have to be address in a separate project that reuses the RichWidgets codebase.

Ref:

[1] https://groups.google.com/d/topic/angular/qG8YRwVaL_g/discussion

**Comment by bleathem** [comment]
Sharing some links from @VineetReynolds:

Bootstrap js for a button wrapped as a widget:
https://github.com/mgcrea/angular-strap/blob/master/src/directives/button.js

Bootstrap js for a button implemented as a native Angular widget:
https://github.com/angular-ui/bootstrap/blob/master/src/buttons/buttons.js

**Comment by bleathem** [comment]
Kendo's approach to building Angular directives:
https://github.com/kendo-labs/angular-kendo/blob/master/src/factories/directive.js

**Comment by pauldijou** [comment]
I totally agree with @VineetReynolds : except for really complex widgets, it would be way better to write AngularJS directives from scratch, taking inspiration from the JS of RichWidgets but doing it the AngularJS way.

It is important to separate model from DOM. This way, you write the logic in the JS of the directive and then do an HTML template, that can be replace by another one if needed. Separation of concernes for the win!

It should be a different project IMO, but sharing as much utils as possible (like all that stuff around offsets to place a tooltip or a popover). But it would require quite some efforts and expertise in the AngularJS framework.
